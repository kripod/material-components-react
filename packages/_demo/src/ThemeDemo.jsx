import React, { PropTypes } from 'react';
import Theme from '../../theme';

const ThemeDemo = props =>
  <div style={{ backgroundColor: '#333' }}>
    <Theme backgroundColor="background">
      {props.children}
    </Theme>

    <Theme color="accent" dark>
      {props.children}
    </Theme>
  </div>
;

ThemeDemo.propTypes = {
  children: PropTypes.node.isRequired,
};

export default ThemeDemo;
