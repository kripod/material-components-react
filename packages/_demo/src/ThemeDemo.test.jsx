import React from 'react';
import renderer from 'react-test-renderer';
import ThemeDemo from './ThemeDemo';

it('renders correctly', () => {
  const tree = renderer.create(<ThemeDemo>Hello, World!</ThemeDemo>).toJSON();
  expect(tree).toMatchSnapshot();
});
