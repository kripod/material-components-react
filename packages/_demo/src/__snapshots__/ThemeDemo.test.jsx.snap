exports[`test renders correctly 1`] = `
<div
  style={
    Object {
      "backgroundColor": "#333",
    }
  }>
  <div
    className="mdc-theme-- mdc-theme--background">
    Hello, World!
  </div>
  <div
    className="mdc-theme--accent mdc-theme---bg mdc-theme--dark">
    Hello, World!
  </div>
</div>
`;
